spec.html

```diff
@@ -6499,31 +6499,31 @@ <h1>GetGlobalObject ( )</h1>
   </emu-clause>
 
   <emu-clause id="sec-jobs-and-job-queues">
-    <h1>Jobs and Host Operations to Enqueue Jobs</h1>
+    <h1>Jobs</h1>
     <p>A <dfn>Job</dfn> is an abstract operation that initiates an ECMAScript computation when no other ECMAScript computation is currently in progress. A Job abstract operation may be defined to accept an arbitrary set of job parameters.</p>
 
-    <p>Jobs are scheduled for execution by ECMAScript host environments with abstract operations such as HostEnqueuePromiseJob. Such operations accept two parameters: _job_, an abstract operation, and _arguments_, a list of ECMAScript values. These abstract operations schedule _job_ to be performed with _arguments_ at some future time. Their implementations must conform to the following requirements:</p>
+    <p>Jobs are scheduled for execution in response to invocations of the abstract operation HostEnqueueJob.</p>
 
-    <ul>
-      <li>At some future point in time, when there is no running execution context and the execution context stack is empty, the implementation must:
-        <ol>
-          <li>Push an execution context onto the execution context stack.</li>
-          <li>Perform any implementation-defined preparation steps.</li>
-          <li>Perform the abstract operation given by _job_, passing the arguments _arguments_.</li>
-          <li>Perform any implementation-defined cleanup steps.</li>
-          <li>Pop the previously-pushed execution context from the execution context stack.</li>
-        </ol>
-      </li>
-      <li>Only one Job may be actively undergoing evaluation at any point in time.</li>
-      <li>Once evaluation of a Job starts, it must run to completion before evaluation of any other Job starts.</li>
-    </ul>
+    <emu-clause id="sec-hostenqueuejob" aoid="HostEnqueueJob">
+      <h1>HostEnqueueJob ( _jobGroup_, _job_, _arguments_ )</h1>
```

Comment:
I prefer to use separate callbacks, rather than passing this string "jobGroup" argument, since in callsites, the string is fixed, and in embedders, the different queues will be processed differently. This phrasing is a bit like serializing the name of the algorithm in a string.

Reply:
> I prefer to use separate callbacks, rather than passing this string "jobGroup" argument

I realize you have that preference, but I don't understand why, and thought that this commit might provide a basis for discussion.

> since in callsites, the string is fixed,

Yup.

> and in embedders, the different queues will be processed differently.

Well, that's up to the host, but in general, yes they will. So? (Do you think those two points are in conflict somehow, or what?)

> This phrasing is a bit like serializing the name of the algorithm in a string.

A bit like that, yes, Conversely, having multiple job-enqueuing operations is a bit like baking the value of the jobGroup parameter into the operation-name. But neither perspective is really a reason for preferring one scheme over the other.

Reply:
> A bit like that, yes, Conversely, having multiple job-enqueuing operations is a bit like baking the value of the jobGroup parameter into the operation-name. But neither perspective is really a reason for preferring one scheme over the other.

I agree that you can argue this both ways. Here, I prefer the current phrasing because the strings tend to be constant, not treated as abstract. We just have to make judgement calls in these cases.

Reply:
> Here, I prefer the current phrasing because the strings tend to be constant, not treated as abstract.

I.e., the first argument to a `HostEnqueueJob` would tend to be a 'string literal' as opposed to a metavariable whose value is a string? Yeah, it wouldn't surprise me if that turned out to be true.